react-app/src/components/ProposalDetailScreen/ProposalDetailScreen.tsx

```diff
@@ -1,34 +1,76 @@
 import React, { useEffect } from "react";
 import cn from "classnames";
-import { Navigate, useParams } from "react-router-dom";
+import { Navigate, useNavigate, useParams } from "react-router-dom";
+import { toast } from "react-toastify";
 import AppRoutes from "../../navigation/AppRoutes";
 import Paper from "../common/Paper/Paper";
-import { RequestStateType } from "../../models/RequestState";
+import {
+  isRequestStateLoaded,
+  RequestStateType,
+} from "../../models/RequestState";
+import { Icon, IconType } from "../common/Icons/Icons";
 import ProposalHeader from "./ProposalHeader";
 import ProposalDescription from "./ProposalDescription";
 import { useProposalQuery } from "./ProposalScreenAPI";
 
 const ProposalDetailScreen: React.FC = () => {
   const { id } = useParams();
   const { fetch, requestState } = useProposalQuery(`proposal_${id}`);
+  const navigate = useNavigate();
 
   useEffect(() => {
     if (id) {
       fetch(id);
     }
   }, [fetch, id]);
 
+  useEffect(() => {
+    switch (requestState.type) {
+      case RequestStateType.Error:
+        toast.error("Failed to fetch proposal.", {
+          toastId: "proposal-detail-request-error",
+        });
+        break;
+      case RequestStateType.Loaded:
+        if (requestState.data === null) {
+          toast.error(`Proposal ${id} does not exist`);
+          navigate(AppRoutes.Proposals);
+        }
+        break;
+      default:
+        console.error("Unrecognized request state type = ", requestState.type);
+        break;
+    }
+  }, [requestState, id, navigate]);
+
```

Comment:
please use `useEffectOnce` to avoid potential infinite loops. See OverviewScreen or ProposalScreen for the usage